src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -430,6 +432,7 @@ public void processRequest(Request request) {
             // the client and leader disagree on where the client is most
             // recently attached (and therefore invalid SESSION MOVED generated)
             cnxn.sendCloseSession();
+            request.checkLatency();
```

Comment:
@eribeiro 
That is exactly the reason that I created my own function after the `cnxn.sendCloseSession()` since I was interested in end to end latency, and didn't use the `zks.serverStats().updateLatency(request.createTime);` method. I would also like to know opinion from others on this. 
Should we scope this JIRA for adding threshold as a general and create a new JIRA for extending the scope to include percentiles with new data-structure inside the `ServerStats` class?